app/routes/cdo/edit/dog-details.js

```diff
@@ -7,67 +7,109 @@ const { getBreeds } = require('../../../api/ddi-index-api/dog-breeds')
 const { addDateComponents } = require('../../../lib/date-helpers')
 const { anyLoggedInUser } = require('../../../auth/permissions')
 const { addBackNavigation, addBackNavigationForErrorCondition, extractBackNavParam } = require('../../../lib/back-helpers')
+const { ApiConflictError } = require('../../../errors/api-conflict-error')
+const Joi = require('joi')
 
-module.exports = [{
-  method: 'GET',
-  path: `${routes.editDogDetails.get}/{indexNumber?}`,
-  options: {
-    auth: { scope: anyLoggedInUser },
-    handler: async (request, h) => {
-      const indexNumber = request.params.indexNumber
-      const dog = await getDogDetails(indexNumber)
-
-      if (dog == null) {
-        return h.response().code(404).takeover()
-      }
-
-      const { breeds } = await getBreeds()
+const mapBoomError = (e, request) => {
+  const { microchipNumber, microchipNumber2 } = request.payload
 
-      if (dog[keys.dateOfBirth]) {
-        addDateComponents(dog, keys.dateOfBirth)
-      }
-      if (dog[keys.dateOfDeath]) {
-        addDateComponents(dog, keys.dateOfDeath)
-      }
-      if (dog[keys.dateExported]) {
-        addDateComponents(dog, keys.dateExported)
-      }
-      if (dog[keys.dateStolen]) {
-        addDateComponents(dog, keys.dateStolen)
-      }
-      if (dog[keys.dateUntraceable]) {
-        addDateComponents(dog, keys.dateUntraceable)
-      }
+  const disallowedMicrochipIds = e.boom.payload.microchipNumbers
 
-      const backNav = addBackNavigation(request)
+  const microchipValidation = Joi.object({
```

Comment:
A bit picky but think should live in the /schema/dog-details.js file rather than inline in the route file